tests/flow/test_operator_precedence.py

```
@@ -0,0 +1,190 @@
+        self.env.assertEqual(result.result_set, [[1]])
+
+    def test05_issue_751_reproduction(self):
+        """Test the exact reproduction case from issue #751"""
+        # Create the node with self-connecting relation as described in the issue
+        self.g.query("CREATE (n0 :L28{k161 : -392104257, k162 : -60652336, k158 : false, id : 5, k159 : 'dGThjm0QF'})")
+        self.g.query("MATCH (n0 {id : 5}), (n1 {id : 5}) MERGE(n0)-[r :T38{k490 : true, k486 : true, k485 : -1782193271, k487 : true, id : 10, k489 : true}]->(n1)")
+        
+        # The problematic query from the issue - should not crash or raise Type mismatch error
+        # WHERE (n1.k159)+'1' CONTAINS '1' should work correctly
+        try:
+            result = self.g.query("MATCH (n0)<-[r0]-(n1), (n2)<-[r1]-(n1), (n0)<-[r2]-(n3) OPTIONAL MATCH (n0)<-[r3]-(n3), (n2)<-[r4]-(n1) WHERE (n1.k159)+'1' CONTAINS '1' AND r3.id <> r4.id AND r4.id <> r3.id AND r3.k485 > r4.k485 AND r4.k485 > r0.k485 RETURN *")
```

Ruff (0.14.10)

Local variable `result` is assigned to but never used

Remove assignment to unused variable `result`

(F841)